src/strongbox/core.clj

```diff
@@ -800,6 +851,39 @@
           (update-installed-addon-list! improved-addon-list)
           (info (format "%s addons checked, %s updates available" num-matched num-updates)))))))
 
+(defn-spec check-for-updates-in-parallel nil?
+  "downloads full details for all installed addons that can be found in summary list"
+  []
+  (when (selected-addon-dir)
+    (let [installed-addon-list (get-state :installed-addon-list)]
+      (when-not (empty? installed-addon-list)
+        (info "checking for updates")
+        (let [queue-atm (get-state :job-queue)
+              check-for-update-job (fn [installed-addon]
+                                     (let [job-id (joblib/addon-id installed-addon)]
+                                       (joblib/create-job-add-to-queue!
+                                        queue-atm
+                                        (partial check-for-update-affective installed-addon)
+                                        {:job-id job-id})))
+
+              ;; create jobs and add them to the queue
+              _ (run! check-for-update-job installed-addon-list)
+
+              expanded-addon-list (try
```

Comment:
remove this try-catch. errors are most likely to come from within the function, that are caught and returned and *not* re-thrown